tx/textual/valuerenderer/dec.go

```diff
@@ -27,5 +29,32 @@ func (vr decValueRenderer) Format(_ context.Context, v protoreflect.Value) ([]Sc
 }
 
 func (vr decValueRenderer) Parse(_ context.Context, screens []Screen) (protoreflect.Value, error) {
-	panic("implement me")
+	if len(screens) != 1 {
+		return nilValue, fmt.Errorf("expected single screen: %v", screens)
+	}
+
+	parsed, err := parseDec(screens[0].Text)
+	if err != nil {
+		return nilValue, err
+	}
+
+	return protoreflect.ValueOfString(parsed), nil
+}
+
+func parseDec(v string) (string, error) {
+	parts := strings.Split(v, ".")
+	if len(parts) > 2 {
+		return "", fmt.Errorf("invalid decimal: too many points in %s", v)
+	}
+
+	intPart, err := parseInt(parts[0])
+	if err != nil {
+		return "", err
+	}
+
+	if len(parts) == 1 {
+		return intPart, nil
+	}
+
+	return intPart + "." + parts[1], nil
```

Comment:
If I pass in `parseDec("1999.😊😊😊😊😊😊😊😊😊😊😊😊😊😊😊")` unfortunately this code is gonna pass. Why couldn't we just strconv.ParseFloat or if we detect no decimal points just use something but at bare minimum we should check that both preDecimal and postDecimal values all have numeric values.